packages/jest-jasmine2/src/index.js

```diff
@@ -69,7 +69,32 @@ function jasmine2(config, environment, moduleLoader, testPath) {
       moduleLoader.requireModule(null, config.setupTestFrameworkScriptFile);
     }
   });
+
+  const iterableEquality = (left, right) => {
+    let result; // result needs to be `undefined` if not an iterator
+    if (left && right && left[Symbol.iterator] && right[Symbol.iterator]) {
+      const leftIterator = left[Symbol.iterator]();
+      const rightIterator = right[Symbol.iterator]();
+      let nextLeft = leftIterator.next();
+      let nextRight = rightIterator.next();
+
+      result = true;
+      while (result && !nextLeft.done) {
```

Comment:
actually, instead of using a while loop, can't you just use a `for of` loop? That way you'll avoid doing the manual iterator advancement for at least one of the iterators.

```
for (const value of a) {
  if (!matches) {
    return false;
  }
  bIterator.next();
}
```